rpc/codec/server.go

```diff
@@ -79,6 +90,43 @@ func (c *serverCodec) ReadRequestHeader(r *rpc.Request) error {
 	return nil
 }
 
+// authenticateRequest takes a request proto and attempts to authenticate it.
+// Requests need to implement RequestWithUser.
+// We compare the header.User against the client certificate Subject.CommonName.
+func (c *serverCodec) authenticateRequest(request proto.Message) error {
+	// RequestWithUser must be implemented.
+	requestWithUser, ok := request.(cockroach_proto.RequestWithUser)
```

Comment:
As mentioned above, you can just define the interface here. Within the `authenticateRequest` function even:

```
  type requestWithUser interface {
    GetUser() string
  }
```